lms/djangoapps/courseware/access.py

```diff
@@ -738,6 +769,50 @@ def _has_staff_access_to_descriptor(user, descriptor, course_key):
     return _has_staff_access_to_location(user, descriptor.location, course_key)
 
 
+def _visible_to_nonstaff_users(obj):
+    """
+    Returns if the object is visible to staff only.
+
+    Arguments:
+        obj: CourseOverview or CourseDescriptor to check
+    """
+    return VisibilityError() if obj.visible_to_staff_only else ACCESS_GRANTED
+
+
+def _has_detached_class_tag(obj):
+    """
+    Returns if the given object has detached as a class tag.
+
+    Arguments:
+        obj: object to check
+    """
+    return ACCESS_GRANTED if 'detached' in obj._class_tags else ACCESS_DENIED  # pylint: disable=protected-access
+
+
+def _has_fulfilled_all_milestones(course_id, user_id):
```

Comment:
Nit: consider keeping the method signature in-sync with the other helper methods.  That is, take in `user, course_id` and then pass through as required.